hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsInputStream.java

```diff
@@ -306,7 +366,7 @@ private int readOneBlock(final byte[] b, final int off, final int len) throws IO
     //If buffer is empty, then fill the buffer.
     if (bCursor == limit) {
       //If EOF, then return -1
-      if (fCursor >= contentLength) {
+      if (fileStatusInformationPresent.get() && fCursor >= getContentLength()) {
```

Comment:
Can you please elaborate on the question please. This would be the case in the first sequential read.

Reply:
my question is that can bcursor already be equal to limit before making the first read ?

Reply:
Yes, for first sequential read, it would be. ref: https://github.com/apache/hadoop/blob/trunk/hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsInputStream.java#L266-L267